ios/storage.md

```diff
@@ -161,317 +161,620 @@ func downloadData() {
 
+```swift
+func application(_ application: UIApplication, handleEventsForBackgroundURLSession identifier: String, completionHandler: @escaping () -> Void) {
+    // Store the completion handler.
+    AWSS3TransferUtility.interceptApplication(application, handleEventsForBackgroundURLSession: identifier, completionHandler: completionHandler)
+}
 ```
-{
-    "version" : "2017-02-28",
-    "operation" : "Query",
-    "query" : {
-        ## Provide a query expression. **
-        "expression": "id = :id",
-        "expressionValues" : {
-            ":id" : {
-                "S" : "${ctx.args.id}"
-            }
-        }
+
+### Managing Transfers When an App Restarts
+
+When an app that has initiated a transfer restarts (if it has been terminated by the system and not force-closed), the transfer may still be in progress or have completed. To make the restarting app aware of the status of transfers, instantiate the transfer utility using the ``AWSS3TransferUtility.s3TransferUtility(forKey: "YOUR_KEY")`` method. AWSS3TransferUtility uses the key to uniquely identify the NSURLSession of the transfers initiated by the app, so it is important to always use the same identifier. AWSS3TransferUtility will automatically reconnect to the transfers that were in progress the last time the app was running.
+
+Though it can be called anywhere in the app, we recommend that you instantiate the AWSS3TransferUtility in the ``appDidFinishLaunching`` lifecycle method. 
+
+### Manage a Transfer when a Suspended App Returns to the Foreground**
+
+When an app that has initiated a transfer becomes suspended and then returns to the foreground, the transfer may still be in progress or may have completed. In both cases, use the following code to re-establish the progress and completion handler blocks of the app.
+
+```swift
+let uploadTasks = transferUtility.getUploadTasks().result
+  for task in uploadTasks! {
+       task.setCompletionHandler(completionHandler!)
+       task.setProgressBlock(progressBlock!)
+}
+
+
+let downloadTasks = transferUtility.getDownloadTasks().result
+   for task in downloadTasks! {
+       task.setCompletionHandler(completionHandler!)
+       task.setProgressBlock(progressBlock!)
     }
 }
+
+
+let multiPartUploadTasks = transferUtility.getMultiPartUploadTasks().result
+    for task in multiPartUploadTasks! {
+        task.setCompletionHandler(completionHandler!)
+        task.setProgressBlock(progressBlock!)
+
+ }
+```
+
+### Transfer with Object Metadata
+
+The `AWSS3TransferUtilityUploadExpression` and `AWSS3TransferUtilityMultiPartUploadExpression` classes contain the method `setValue:forRequestHeader` where you can pass in metadata to Amazon S3. This example demonstrates passing in the Server-side Encryption Algorithm as a request header in uploading data to S3 using MultiPart.
```

Comment:
We should link to official S3 docs here: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html

Reply:
Done